extra/migrate_settings.rb

```diff
@@ -1,108 +1,228 @@
+      raise "Migration state file #{@migration_state_file_path} can't be found" unless File.exist?(@migration_state_file_path)
+      @past_migrations = File.read(@migration_state_file_path).split("\n")
+      self
+    end
+
+    def register_migration(migration_class)
+      @migrations ||= []
+      @migrations << migration_class
+    end
+
+    def new_migrations
+      # don't want to deal with namespaced constant resolution/loading
+      migrations.select {|m| !past_migrations.include?(m.name) }
+    end
+
+    def persist_migrations_state(result_dir_path)
+      migration_state = File.open(File.join(result_dir_path, "migration_state"), "w")
+      migration_state.write((past_migrations + executed_migrations.map {|m| m.class.name}).uniq.join("\n"))
+    ensure
+      migration_state.close unless migration_state.nil?
+    end
+  end
+
+  class Migration
+    attr_reader :working_dir_path
+
+    class << self
+      def inject_migrations_instance(migrations)
+        @@migrations = migrations
+      end
+
+      def inherited(subclass)
+        @@migrations.register_migration(subclass)
       end
     end
+
+    def initialize(working_dir_path)
+      @working_dir_path = working_dir_path
+    end
+
+    def migration_name
+      underscore(self.class.name)
+    end
+
+    def migration_dir
+      File.join(working_dir_path, migration_name)
+    end
+
+    def src_dir
+      File.join(migration_dir, "src")
+    end
+
+    def dst_dir
+      File.join(migration_dir, "dst")
+    end
+
+    def underscore(src)
+      src = src.gsub(/::/, '/')
+      src = src.gsub(/([A-Z]+)([A-Z][a-z])/,'\1_\2')
+      src = src.gsub(/([a-z\d])([A-Z])/,'\1_\2')
+      src = src.tr("-", "_")
+      src.downcase
+    end
   end
 
-  # Rename the port to whichever is correct
-  if output[:settings].keys.include?(:port)
-    if output[:settings].keys.include?(:ssl_certificate)
-      output[:settings][:https_port] = output[:settings].delete(:port)
-    else
-      output[:settings][:http_port] = output[:settings].delete(:port)
+  class Migrator
+    attr_reader :working_dir_path, :migrations_dir_path, :config_src_path, :modules_config_src_path, :result_dir_path
+
+    def initialize(working_dir_path, migrations_dir_path, config_src_path, modules_config_src_path, migrations)
+      @working_dir_path = working_dir_path
+      @migrations_dir_path = migrations_dir_path
+      @config_src_path = config_src_path
+      @modules_config_src_path = modules_config_src_path
+      @migrations = migrations
+      @result_dir_path = File.join(working_dir_path, 'result')
+    end
+
+    def load_migrations(migrations_dir_path)
+      migration_files = Dir.glob(File.join(migrations_dir_path, "*.rb")).sort
+      ::Proxy::Migration.inject_migrations_instance(@migrations)
+      migration_files.each { |file| require file }
+    end
+
+    def migrate
+      raise "Migrations dir '#{migrations_dir_path}' doesn't exist" unless Dir.exist?(migrations_dir_path)
```

Comment:
`Dir.exist?` is throwing an error on Ruby 1.8.7, suggest you perhaps use `File.directory?`